### DISCUSSION TITLE
chapter_tree/summary/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#122] [Announcements]
# chapter_tree/summary/

一本动画图解、能运行、可提问的数据结构与算法入门书

https://www.hello-algo.com/chapter_tree/summary/

**Comment by yyym2017** [comment]
# 内容还是太少了，跟考研的要求相差甚远

**Comment by onble** [comment]
毕业工作一年了，看到这里感觉对树结构有了进一步的认识。
大二学数据结构时，学到深度遍历，广度遍历，感觉这两种算法单纯为了学而学，不知道有什么用。
后来陆续接触到了几个树结构的事物：1.python有一个方向是爬虫，当爬取网页时，深度优先和广度优先获取数据是有区别的。
2.网页的组成就是document树结构，当爬取下来信息，从中抽取文本内容时，也需要考虑如何遍历document树结构。
3.深度优先和广度优先的思想用在了A*寻路算法（印象中A*也可以不用，记不清了=-=）
总结：树结构在生活中还是有挺多例子的，不是像高数那样纯抽象来做筛选的（高数真不如线性代数好用，游戏领域对线性代数有好多的应用）。
